Firmware/ultralcd.cpp

```diff
@@ -2814,7 +2813,7 @@ bool lcd_calibrate_z_end_stop_manual(bool only_z)
             }
         }
         // Let the user confirm, that the Z carriage is at the top end stoppers.
-        uint8_t result = lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Are left and right Z~carriages all up?"), false);////MSG_CONFIRM_CARRIAGE_AT_THE_TOP c=20 r=3
+        uint8_t result = lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Are left and right Z-carriages all up?"), false);////MSG_CONFIRM_CARRIAGE_AT_THE_TOP c=20 r=3
```

Comment:
`~` is usually parsed in the full screen code as "white space". We need to test this message just to be sure it still renders OK.

See in `lcd_display_message_fullscreen_nonBlocking_P`:

```
if (c == '~')
                c = ' ';
```

Reply:
Oh damn. I didn't know that was the case. We should definitely check this message. If it renders fine, then I propose we remove this stupid exception.

Reply:
Hmm I just noticed this is the only message that was using `~` like this? Seems like a low hanging fruit to remove it to simplify the code. I thought this was used in more messages.

Reply:
Some translations may need to be updated. Like this one:

```
"Calibrating XYZ. Rotate the knob to move the Z carriage up to the end "
"stoppers. Click when done."
msgstr ""
"Kalibrace XYZ. Otacenim tlacitka posunte Z osu az k~hornimu dorazu. Potvrdte"
" tlacitkem."
```

Reply:
@gudnimg I see what this is supposed to do. It's used to enforce that `Z` and `carriage` are not split between lines. This is stupid. A dash between the words is just as effective and it doesn't require this stupid hack. Do you know if any other message makes use of this feature?

Reply:
It seems to be a bit inconsistent between translations. Some use `~` and some don't.
I found this one as well:

```
#. MSG_WIZARD_WELCOME_SHIPPING c=20 r=12
#: ../../Firmware/messages.cpp:126 ../../Firmware/ultralcd.cpp:3803
msgid ""
"Hi, I am your Original Prusa i3 printer. I will guide you through a short "
"setup process, in which the Z-axis will be calibrated. Then, you will be "
"ready to print."
msgstr ""
"Bok, ja sam vas Original Prusa i3 printer. Vodit cu vas kroz kratak proces "
"postavljanja, u kojem ce se kalibrirati Z~os. Tada cete biti spremni za "
"printanje."
```

Reply:
I pushed a commit. I think I removed all tildes. We need to check how the text looks now. Worst case, we enforce a "page" break in the message so that words split between lines better.